packages/amplify_analytics_pinpoint/ios/Classes/FlutterAnalytics.swift

```diff
@@ -40,8 +39,23 @@ public class FlutterAnalytics {
                         "recoverySuggestion" : analyticsError.recoverySuggestion,
                         "underlyingError": analyticsError.underlyingError != nil ? analyticsError.underlyingError!.localizedDescription : ""
                     ])
-            }
-            else{
+            } else if(error is ConfigurationError) {
+                let configError = error as! ConfigurationError
+                
+                var errorCode = "AnalyticsException"
+                if case .amplifyAlreadyConfigured = configError {
+                   errorCode = "AmplifyAlreadyConfiguredException"
+                }
+                ErrorUtil.postErrorToFlutterChannel(
+                    result: result,
+                    errorCode: errorCode,
+                    details: [
+                        "message" : configError.errorDescription,
+                        "recoverySuggestion" : configError.recoverySuggestion,
+                        "underlyingError": configError.underlyingError != nil ? configError.underlyingError!.localizedDescription : ""
+                    ]
+                )
+            } else{
```

Comment:
Can this whole block be extracted as a common util?

Reply:
Yes it can! I'm going to do this, if I have time.

Reply:
Yeah that's a great point.  Could the code even live in the Core package so that it is not rewritten for every plugin?